### PR TITLE
KafkaSource reconciler tests for ConsumerGroup scaling

### PR DESCRIPTION
**Pull request by aavarghese**
Signed-off-by: aavarghese <avarghese@us.ibm.com>

Part of #1537
~Will be rebased on #1754~ **done**

/cc @pierDipi 

## Proposed Changes

- Adding new KafkaSource reconciler unit tests for consumer scheduling and scaling

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/source/v2/sourcev2.go [hunk 1]
```diff
@@ -64,11 +65,15 @@ func (r *ReconcilerV2) ReconcileKind(ctx context.Context, ks *sources.KafkaSourc
...
-			topLevelCondition.Message,
-		)
+		if topLevelCondition == nil {
+			ks.GetConditionSet().Manage(&ks.Status).MarkUnknown(KafkaConditionConsumerGroup, "failed to reconcile consumer group", "consumer group not ready")
+		} else {
+			ks.GetConditionSet().Manage(&ks.Status).MarkFalse(
+				KafkaConditionConsumerGroup,
+				topLevelCondition.Reason,
+				topLevelCondition.Message,
+			)
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/source/v2/sourcev2.go:75]: Can we add another test to verify that the top-level condition is propagated?

**Review by pierDipi — COMMENTED** [review]
Thanks, LGTM, left a comment.

File: control-plane/pkg/reconciler/source/v2/sourcev2.go (hunk 1 above)
Comment (aavarghese) [reply control-plane/pkg/reconciler/source/v2/sourcev2.go:75]: Yes, adding one similar to trigger's...

**Review by pierDipi — APPROVED** [review]
/lgtm
/retest

**Pull request merged**